src/libraries/Microsoft.PowerFx.Core/Glue/SimpleResolver.cs

```diff
@@ -48,27 +51,44 @@ internal class SimpleResolver : INameResolver
 
         public IExternalEntity CurrentEntity => null;
 
-        public SimpleResolver(IEnumerable<EnumSymbol> enumSymbols, params TexlFunction[] extraFunctions)
-            : this(extraFunctions)
+        /// <summary>
+        /// Initializes a new instance of the <see cref="SimpleResolver"/> class.
+        /// </summary>
+        /// <param name="config"></param>
+        public SimpleResolver(PowerFxConfig config)            
         {
-            _enums = enumSymbols.ToArray();
-        }
+            _config = config ?? throw new ArgumentNullException(nameof(config));
 
-        public SimpleResolver(params TexlFunction[] extraFunctions)
-        {
-            _library = GetFullFunctionLibrary(extraFunctions);
-        }
-
-        public static TexlFunction[] GetFullFunctionLibrary(TexlFunction[] extraFunctions)
-        {
-            var list = new List<TexlFunction>();
-            list.AddRange(BuiltinFunctionsCore.BuiltinFunctionsLibrary);
-            list.AddRange(extraFunctions);
-            return list.ToArray();
+            _library = config.Functions.ToArray();
+            _enums = config.EnumStore.WithRequiredEnums(_library).EnumSymbols.ToArray();
         }
-
+        
         public virtual bool Lookup(DName name, out NameLookupInfo nameInfo, NameLookupPreferences preferences = NameLookupPreferences.None)
         {
+            if (_config != null)
+            {
+                if (_config.TryGetSymbol(name, out var symbol, out var displayName))
```

Comment:
Why does _config have a field called TryGetSymbol? Would it be better if PowerFx config implemented INameResolver directly?

Reply:
INameResolve has many different resolve hooks. 

"Would it be better if PowerFx config implemented INameResolver directly" - this PR moves us *much* closer to that possibility.  
However, there are still derived cases that need to change resolution in a way that the config doesn't handle yte.